doc/tutorials/crypto.md

```diff
@@ -0,0 +1,275 @@
+# Cryptographic Methods in Elektra
+
+Elektra can protect the following aspects of your configuration:
+
+1. confidentiality (i.e. protection against unauthorized access), and
+2. integrity (i.e. protection against unauthorized modification).
+
+Elektra provides two plugins to achieve this protection:
+
+1. `crypto`, and
+2. `fcrypt`.
+
+## Prerequisites - GnuPG
+
+For the rest of this tutorial we assume that you are somewhat familiar with GnuPG (GPG).
+The documentation of GnuPG can be found [here](https://gnupg.org/documentation/index.html).
+
+In order to find your GPG private key(s) you can use:
+
+	gpg2 --list-secret-keys
+
+If GPG private keys are available, you see an output, that looks similar to this:
+
+	sec   rsa1024 2016-08-20 [SC]
+		  DDEBEF9EE2DC931701338212DAF635B17F230E8D
+	uid           [ultimate] Elektra Unit Tests (DO NOT USE IN PRODUCTION) <unit-tests@libelektra.org>
+	ssb   rsa1024 2016-08-20 [E]
+
+The GPG key we use in this tutorial has the ID `DDEBEF9EE2DC931701338212DAF635B17F230E8D`.
+
+A GPG private key is mandatory for the plugins to work.
+If you have no GPG private key available, you can generate one by entering the following command:
+
+	gpg2 --generate-key
+
+The `fcrypt` plugin and the `crypto` plugin support both versions (version 1 and version 2) of GPG.
+
+In order to set up our tutorial we import the Elektra test key.
+We **DO NOT RECOMMEND** to use our key on your local machine, as it is available to the public!
+
+```sh
+gpg2 --import src/plugins/crypto/test_key.asc || gpg --import src/plugins/crypto/test_key.asc
+echo "trust-model always" > ~/.gnupg/gpg.conf
```

Comment:
I think this line is quite dangerous. This command just overwrote my local GPG configuration.